### PR TITLE
Remove source link in Findings Report when source is none

### LINKED ISSUE #3636: Findings report - Source none has link
**Describe the bug**
When creating a Findings Report, when a source is None, it still contains a href, which gives an error when you click on it.

**To Reproduce**

1. Create a findings report
2. Expand the row
3. Click on the link of the "None" under Source

**Expected behavior**
When there is no source, there shouldn't be a href.

**Screenshots**
Image

**OpenKAT version**
main

### PR DESCRIPTION
**Pull request by madelondohmen**
### Changes
This PR removes the link from the Occurrence's source when the source is None.

### Issue link
Closes #3636

### Demo
Before:
afbeelding

After:
afbeelding

### QA notes
- Create a Findings Report (one with a KAT-NAMESERVER-NO-IPV6 or KAT-NO-DNSSEC finding for example)
- Expand the table rows and check for a source == None
- The link should be removed
- The link for sources that are not None should still be visible and working

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/reports/report_types/findings_report/report.html [hunk 1]
```diff
@@ -50,7 +50,11 @@ <h3>{% translate "Findings" %}</h3>
                                 <h4 class="heading-normal">{% translate "Description" %}</h4>
                                 <p>{{ info.finding_type.description }}</p>
                                 <h4 class="heading-normal">{% translate "Source" %}</h4>
-                                <a href="{{ info.finding_type.source }}">{{ info.finding_type.source }}</a>
+                                {% if info.finding_type.source %}
+                                    <a href="{{ info.finding_type.source }}">{{ info.finding_type.source }}</a>
+                                {% else %}
+                                    <p>{{ info.finding_type.source }}</p>
+                                {% endif %}
```
Comment (underdarknl) [inline rocky/reports/report_types/findings_report/report.html:57]: Does it make sense to even print None or Source here at all if there is no source?

Comment (madelondohmen) [reply rocky/reports/report_types/findings_report/report.html:57]: If we decide not to do this, we should also remove the option to display None from all other headings (Description, Impact and Recommendation).

**Review by ammar92 — APPROVED** [review]

**Comment by underdarknl** [comment]
Could we validate that this Finding does move from Pending to hydrated given enough time?

**Comment by stephanie0x00** [comment]
Verified this, adding the KAT-NAMESERVER-NO-IPV6 finding manually resolves into a Recommendation finding. The pending is cleared.

**Pull request merged**